Mathlib/CategoryTheory/Abelian/GrothendieckCategory.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2024 Paul Reichert. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+
+import Mathlib.CategoryTheory.Abelian.Subobject
+import Mathlib.CategoryTheory.Abelian.GrothendieckAxioms
+import Mathlib.CategoryTheory.Adjunction.AdjointFunctorTheorems
+
+/-!
+
+# Grothendieck categories
+
+This file defines Grothendieck categories and proves basic facts about them.
+
+## Definitions
+
+A `GrothendieckCategory` is an abelian category provided that it has `AB5` and a separator.
+
+## Theorems
+
+Relevant implications of `GrothendieckCategory` are established in `GrothendieckCategory.hasLimits`
+and `GrothendieckCategory.hasColimits`.
+
+## References
+
+* [Stacks: Grothendieck's AB conditions](https://stacks.math.columbia.edu/tag/079A)
+
+-/
+
+namespace CategoryTheory
+
+open Limits
+
+universe u v
+variable (C : Type u) [Category.{v} C]
+
+/--
+An abelian category `C` is called a Grothendieck category provided that it has `AB5` and a
+separator (see `HasSeparator`).
+-/
+@[stacks 079B]
+class GrothendieckCategory [Abelian C] where
```

Comment:
When #19200 is merged, I would suggest you use the following definition:
```lean
-- universe w v u
class IsGrothendieckAbelian : Prop where
  locallySmall : LocallySmall.{w} C := by infer_instance
  hasFilteredColimitsOfSize : HasFilteredColimitsOfSize.{w, w} C := by infer_instance
  ab5OfSize : AB5OfSize.{w, w} C := by infer_instance
  hasSeparator : HasSeparator C := by infer_instance
```
Here, there is an auxiliary universe `w`. The first thing to prove would be that under `IsGrothendieckAbelian.{w} C`, we can prove ` IsGrothendieckAbelian.{w, w} (ShrinkHoms C) `. The idea is that under certain circumstances (e.g. étale sheaves), the morphisms in `C` (which are in `Type v`) shall we `w`-small (for a small universe), and we would like to get the good properties of Grothendieck abelian categories for such categories. Roughly speaking, a category satisfying `IsGrothendieckAbelian.{w}` is a category that is equivalent to a category (with `Category.{w}`) that is Grothendieck abelian for your definition.
In order to develop the API, it may be convenient to prove first some results under the favourable assumption `w = v`, and then we may transport properties using the equivalence of categories `ShrinkHoms.equivalence.{w} C`.